sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -441,6 +441,7 @@ object FunctionRegistry {
     // collection functions
     expression[CreateArray]("array"),
     expression[ArrayContains]("array_contains"),
+    expression[HasAll]("has_all"),
```

Comment:
The naming might be controversial later.

Reply:
> The naming might be controversial later.

Do you have other idea about the naming.
As we have discussed in https://github.com/apache/spark/pull/30243#discussion_r532255830

Reply:
Ya. I saw the discussion history, but the problem is that it doesn't come from the popular DBMSes.

Reply:
> Ya. I saw the discussion history, but the problem is that it doesn't come from the popular DBMSes.

Firbase have similar function also called such way.
https://firebase.google.com/docs/reference/rules/rules.List#hasAll

Popular DBMS such as mysql, sql server, oracle have less array functions and PostgresSQL's naming is strange...

Reply:
Any more concern ? Popular DBMS don't have less high-level  functions since slow iteration.
And this method is really useful when many business scenario